x-pack/plugins/alerting/README.md

```diff
@@ -0,0 +1,232 @@
+|---|---|---|
+|alertTypeId|The id value of the alert type you want to call when the alert is scheduled to execute.|string|
+|interval|The interval in milliseconds the alert should execute.|number|
+|alertTypeParams|The parameters to pass in to the alert type executor `params` value.|object|
+|action|An array of `group` (string), `id` (string) and params (object) to fire whenever the alert fires. The group allows the alert type fire fire different groups of actions. For example `warning` and `severe`. The id is the id of the action saved object to use. The params are the `params` value the action type expects. This uses mustache templates recursively on strings, see templating actions section for more details.|array|
+
+#### `DELETE /api/alert/{id}`: Delete alert
+
+Params:
+
+|Property|Description|Type|
+|---|---|---|
+|id|The id of the alert you're trying to delete.|string|
+
+#### `GET /api/alert/_find`: Find alerts
+
+Params:
+
+See saved objects API documentation for find, all the properties are the same except you cannot pass in `type`.
+
+#### `GET /api/alert/{id}`: Get alert
+
+Params:
+
+|Property|Description|Type|
+|---|---|---|
+|id|The id of the alert you're trying to get.|string|
+
+#### `GET /api/alert/types`: List alert types
+
+No parameters.
+
+#### `PUT /api/alert/{id}`: Update alert
+
+Params:
+
+|Property|Description|Type|
+|---|---|---|
+|id|The id of the alert you're trying to update.|string|
+
+Payload:
+
+|Property|Description|Type|
+|---|---|---|
+|interval|The interval in milliseconds the alert should execute.|number|
+|alertTypeParams|The parameters to pass in to the alert type executor `params` value.|object|
+|action|An array of `group` (string), `id` (string) and params (object) to fire whenever the alert fires. The group allows the alert type fire fire different groups of actions. For example `warning` and `severe`. The id is the id of the action saved object to use. The params are the `params` value the action type expects. This uses mustache templates recursively on strings. The templates have access to `context` and `state` objects.|array|
+
+## Alert instance factory
+
+**alertInstanceFactory(id)**
+
+One service passed in to alert types is an alert instance factory. This factory creates instances of alerts and lookups of previous instances with the same id to gather previous state.
+
+This returns an instance of `AlertInstance`. The alert instance class has the following methods, note that most of these are for internal use but they work as they are described:
+
+|Method|Description|
+|---|---|
+|shouldFire()|Function returns `true` or `false` whether the alert instance should fire or not. This value turns to true when `.fire(...)` is called.|
+|getFireOptions()|Function returns an object of `actionGroup`, `context` and `state` to use for firing the actions.|
+|resetFire()|Reverts a `.fire(...)` call and makes the instance no longer fire.|
+|getState()|Get the current state of the alert instance.|
```

Comment:
Yeah I'll try to rethink this one, its hard to explain what it will contain since the developer is in control of that. But I can mark where the values will come from / be set from.